src/kinetics/Reaction.cpp

```diff
@@ -891,6 +825,329 @@ void BlowersMaselInterfaceReaction::getParameters(AnyMap& reactionNode) const
+                node["equation"].asString());
+        }
+        return;
+    }
+
+    reactants.erase("M");
+    products.erase("M");
+    m_third_body->setEfficiencies(node);
+}
+
+void ThreeBodyReaction3::getParameters(AnyMap& reactionNode) const
+{
+    ElementaryReaction3::getParameters(reactionNode);
+    if (!specified_collision_partner) {
+        reactionNode["type"] = "three-body";
+        reactionNode["efficiencies"] = m_third_body->efficiencies;
+        reactionNode["efficiencies"].setFlowStyle();
+        if (m_third_body->default_efficiency != 1.0) {
+            reactionNode["default-efficiency"] = m_third_body->default_efficiency;
+        }
+    }
+}
+
+std::string ThreeBodyReaction3::reactantString() const
+{
+    if (specified_collision_partner) {
+        return ElementaryReaction3::reactantString() + " + "
+            + m_third_body->efficiencies.begin()->first;
+    } else {
+        return ElementaryReaction3::reactantString() + " + M";
+    }
+}
+
+std::string ThreeBodyReaction3::productString() const
+{
+    if (specified_collision_partner) {
+        return ElementaryReaction3::productString() + " + "
+            + m_third_body->efficiencies.begin()->first;
+    } else {
+        return ElementaryReaction3::productString() + " + M";
+    }
+}
+
+PlogReaction3::PlogReaction3()
+    : Reaction3()
+{
+    m_rate = std::shared_ptr<PlogRate>(new PlogRate);
+}
+
+PlogReaction3::PlogReaction3(const Composition& reactants,
+                             const Composition& products, const PlogRate& rate)
+    : Reaction3(reactants, products)
+{
+    m_rate = std::make_shared<PlogRate>(rate);
+}
+
+PlogReaction3::PlogReaction3(const AnyMap& node, const Kinetics& kin)
+    : PlogReaction3()
+{
+    setParameters(node, kin);
+    setRate(std::shared_ptr<PlogRate>(new PlogRate(node, rate_units)));
+}
+
+void PlogReaction3::getParameters(AnyMap& reactionNode) const
+{
+    Reaction::getParameters(reactionNode);
+    reactionNode["type"] = "pressure-dependent-Arrhenius";
+    reactionNode.update(m_rate->parameters(rate_units));
+}
+
+ChebyshevReaction3::ChebyshevReaction3()
+    : Reaction3()
+{
+    m_rate = std::shared_ptr<ChebyshevRate3>(new ChebyshevRate3);
+}
+
+ChebyshevReaction3::ChebyshevReaction3(const Composition& reactants,
+                                       const Composition& products,
+                                       const ChebyshevRate3& rate)
+    : Reaction3(reactants, products)
+{
+    m_rate = std::make_shared<ChebyshevRate3>(rate);
+}
+
+ChebyshevReaction3::ChebyshevReaction3(const AnyMap& node, const Kinetics& kin)
+    : ChebyshevReaction3()
+{
+    setParameters(node, kin);
+    setRate(std::shared_ptr<ChebyshevRate3>(new ChebyshevRate3(node, rate_units)));
+}
+
+void ChebyshevReaction3::setParameters(const AnyMap& node, const Kinetics& kin)
+{
+    if (!node.hasKey("equation")) {
+        // empty node: used by newReaction() factory loader
+        return;
+    }
```

Comment:
I guess it would be possible to check for the existence of a non-empty `m_third_body`, but depending on when this is called, it may not be detected yet. Overall, I prefer to use an explicit overload here as it is much simpler to understand later on, especially as this is presumably not a performance-limiting method.

Reply:
Merged with `Reaction::setParameters`.